src/main/java/edu/project2/SolutionMaze.java

```diff
@@ -0,0 +1,99 @@
+package edu.project2;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class SolutionMaze {
+    private static final String FAILED_SOLUTION = "Failed to solve maze.";
+
+    private SolutionMaze() {
+    }
+
+    public static void solveDFS(Maze maze) {
+        boolean[][] visited = new boolean[maze.getHeight()][maze.getWeight()];
+        for (int i = 0; i < maze.getHeight(); i++) {
+            for (int j = 0; j < maze.getWeight(); j++) {
+                visited[i][j] = false;
+            }
+        }
+
+        if (!dfs(maze.getStart(), maze.getEnd(), maze, visited)) {
```

Comment:
С точки зрения читаемости кода лучше явно вынести результат решения в переменную:
```suggestion
        boolean isSolved = dfs(maze.getStart(), maze.getEnd(), maze, visited)
        if (!isSolved) {
```
чтобы было сразу понятно, где именно происходит решение лабиринта.